modules/runner/src/integrationTest/java/org/apache/ignite/internal/sql/engine/ItUuidTest.java

```diff
@@ -265,4 +266,22 @@ private void checkUuidComparison(Object uuid1, Object uuid2) {
         assertQuery("SELECT ? < ?").withParams(uuid1, uuid2).returns(false).check();
         assertQuery("SELECT ? <= ?").withParams(uuid1, uuid2).returns(false).check();
     }
+
+    @Test
+    public void testRandomUuid() {
+        assertQuery("INSERT INTO t SELECT x, RAND_UUID(), RAND_UUID() FROM TABLE(SYSTEM_RANGE(0, 99))").returns(100L).check();
+        assertQuery("SELECT uuid_key FROM t").columnTypes(UUID.class).check();
+        assertQuery("SELECT COUNT(*) FROM t WHERE uuid_key = uuid_key2").returns(0L).check();
+        assertQuery("SELECT COUNT(*) FROM t WHERE uuid_key = RAND_UUID()").returns(0L).check();
+        assertQuery("SELECT COUNT(*) FROM t WHERE uuid_key != uuid_key2").returns(100L).check();
+    }
+
+    @Test
+    @Disabled("https://issues.apache.org/jira/browse/IGNITE-18931")
+    public void testRandomUuidComparison() {
+        assertQuery("SELECT RAND_UUID() = RAND_UUID()").returns(false).check();
+        assertQuery("SELECT RAND_UUID() != RAND_UUID()").returns(true).check();
+        assertQuery("SELECT RAND_UUID() >= RAND_UUID()").returns(true).check();
+        assertQuery("SELECT RAND_UUID() <= RAND_UUID()").returns(true).check();
```

Comment:
These checks will passed only if "RAND_UUID() = RAND_UUID()",
but you expect ""SELECT RAND_UUID() = RAND_UUID()" returns 'false' above.

Reply:
If you just want to check if RAND_UUID can participate in inequality, the try smth like this:
"SELECT COUNT(*) FROM t WHERE uuid_key > RAND_UUID or uuid_key < RAND_UUID"

Reply:
Thanks, I just wanted to add checks that ">=" and "<=" comparisons would be correctly optimized, but the tests themselves are wrong, so I removed them.

Reply:
`WHERE uuid_key > RAND_UUID or uuid_key < RAND_UUID`
is the same as
`WHERE uuid_key != RAND_UUID`
but uses range checks.

Reply:
Maybe I don't quite understand or something is not working right.
1. currently each call to RAND_UUID will generate different value
2. UUID is comparable

So `SELECT COUNT(*) FROM t WHERE uuid_key > RAND_UUID or uuid_key < RAND_UUID` 
should return from 0 to 100% of the number of rows.